### PR TITLE
Add ability to move up/down in list

### LINKED ISSUE #23: Add ability to move up/down in list
Sometimes you don't really care to reorder the entire list but just one or two entities, and having a helper method for this would be useful.

### PR DESCRIPTION
**Pull request by josegonzalez**
I don't believe in tests so I'll leave that as an exercise to the merger.

Closes #23.

### CODE REVIEW

File: src/Model/Behavior/SequenceBehavior.php [hunk 1]
```diff
@@ -259,6 +246,89 @@ public function beforeDelete(Event $event, Entity $entity)
...
 
     /**
+     * Decrease the position of the entity on the list
+     *
+     * If a "higher" entity exists, this will also swap positions with it
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity that is going to be saved.
+     * @return bool
+     */
+    public function moveUp(EntityInterface $entity)
```
Comment (ADmad) [inline src/Model/Behavior/SequenceBehavior.php]: Can add a second param `$number` similar to that of [TreeBehavior::moveUp()](https://api.cakephp.org/3.4/class-Cake.ORM.Behavior.TreeBehavior.html#_moveUp). Same for the `moveDown()` method.

Comment (josegonzalez) [inline src/Model/Behavior/SequenceBehavior.php]: Then I'd have to worry about shifting more entities around, is that fine by you?

Comment (ADmad) [inline src/Model/Behavior/SequenceBehavior.php]: Hmm.. If it makes the code too complicated leave it be :)

**Comment by ADmad** [comment]
> I don't believe in tests so I'll leave that as an exercise to the merger.

At least ensure existing tests pass, which don't right now :)

**Comment by josegonzalez** [comment]
Okay tests pass again, now you just need to add coverage for this :D

File: src/Model/Behavior/SequenceBehavior.php (hunk 1 above)
Comment (josegonzalez) [inline src/Model/Behavior/SequenceBehavior.php]: Yeah I don't think that's quite necessary, and then lends itself to features like "move all the way to top/bottom". Maybe as a separate PR, but this code is probably more useful than the other features.

**Comment by bravo-kernel** [comment]
> I don't believe in tests so I'll leave that as an exercise to the merger.

Priceless 🤣

**Comment by ADmad** [comment]
@bravo-kernel It's no surprise then that the code was non functional. The patch has gems like `return _movePosition($entity, $direction = '+');` instead of `return $this->_movePosition($entity, $direction = '+');` :stuck_out_tongue_closed_eyes:

**Comment by bravo-kernel** [comment]
I can't believe I am saying this but that's exactly why I appreciate @jadb's insane method of only adding tests better (nowadays), leaving the implementation to the merger :trollface:

**Comment by ADmad** [comment]
> ... why I appreciate @jadb's insane method of only adding tests better (nowadays), leaving the implementation to the merger

True TDD :)

**Comment by josegonzalez** [comment]
Close enough.

**Pull request merged**